modules/PersistentStore/main/kvsImpl.cpp

```diff
@@ -0,0 +1,234 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 - 2022 Pionix GmbH and Contributors to EVerest
+
+#include "kvsImpl.hpp"
+
+#include <boost/filesystem.hpp>
+#include <boost/variant.hpp>
+
+namespace module {
+namespace main {
+
+void kvsImpl::init() {
+    // open and initialize database
+    boost::filesystem::path sqlite_db_path =
+        boost::filesystem::absolute(boost::filesystem::path(mod->config.sqlite_db_file_path));
+    boost::filesystem::path database_directory = sqlite_db_path.parent_path();
+    if (!boost::filesystem::exists(database_directory)) {
+        boost::filesystem::create_directories(database_directory);
+    }
+
+    int ret = sqlite3_open(sqlite_db_path.c_str(), &this->db);
+
+    if (ret != SQLITE_OK) {
+        EVLOG(error) << "Error opening PersistentStore database '" << sqlite_db_path << "': " << sqlite3_errmsg(db);
+        throw std::runtime_error("Could not open PersistentStore database at provided path.");
+    }
+
+    EVLOG(debug) << "Using SQLite version " << sqlite3_libversion();
+
+    // prepare the database
+    std::string create_sql = "CREATE TABLE IF NOT EXISTS KVS ("
+                             "KEY   TEXT UNIQUE,"
+                             "VALUE TEXT,"
+                             "TYPE  TEXT);";
+
+    sqlite3_stmt* create_statement;
+    sqlite3_prepare_v2(this->db, create_sql.c_str(), create_sql.size(), &create_statement, NULL);
+    int res = sqlite3_step(create_statement);
+    if (res != SQLITE_DONE) {
+        EVLOG(error) << "Could not create KVS table: " << res << sqlite3_errmsg(this->db);
+        throw std::runtime_error("PersistentStore db access error");
+    }
+
+    if (sqlite3_finalize(create_statement) != SQLITE_OK) {
+        EVLOG(error) << "Error creating KVS table";
+        throw std::runtime_error("PersistentStore db access error");
+    }
+}
+
+void kvsImpl::ready() {
+}
+
+class TypeNameVisitor : public boost::static_visitor<std::string> {
```

Comment:
I would have put both visitors into one, returning a struct with containing type and text